marathon/src/test/scala/io/buoyant/marathon/v2/ApiTest.scala

```diff
@@ -56,16 +56,37 @@ class ApiTest extends FunSuite with Awaits with Exceptions {
 
   val appNotFoundBuf = Buf.Utf8("""{"message":"App '/foo' does not exist"}""")
 
+  val authbuf = Buf.Utf8("""{"token":"foo"}""")
+  val auth = Some(Api.Auth("/fakeauth", "fakecontent"))
+
   def stubService(buf: Buf) = Service.mk[Request, Response] { req =>
     val rsp = Response()
     rsp.content = buf
     Future.value(rsp)
   }
 
+  case class stubAuth(authbuf: Buf, buf: Buf) {
+    var authed = false
+    def service() =
+      Service.mk[Request, Response] { req =>
+        val rsp = Response()
+        if (!authed) {
```

Comment:
what about conditioning on the path?  If the path is the authPath, serve the authBuf.  otherwise assert that the request has a valid token and serve the buf.